### PR TITLE
Support Spotify

### PR DESCRIPTION
**Pull request by zjgcjy**
see https://github.com/yihong0618/GitHubPoster/issues/67

### CODE REVIEW

File: github_poster/loader/config.py [hunk 1]
```diff
@@ -125,3 +125,9 @@
 
 # COVID
 COVID_API = "https://pomber.github.io/covid19/timeseries.json"
+
+# Spotify
+SPOTIFY_TOKEN = "https://accounts.spotify.com/api/token"
```
Comment (yihong0618) [inline github_poster/loader/config.py:130]: SPOTIFY_TOKEN_URL or SPOTIFY_TOKEN_API

File: github_poster/loader/spotify_loader.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        )
+
+    def _parse_spotify_history(self):
+        if os.path.exists(self.spotify_file):
+            with open(self.spotify_file, "r") as f:
+                self.number_by_date_dict = json.load(f)
+
+    def _writeback_spotify_history(self):
+        with open(self.spotify_file, "w") as f:
+            json.dump(self.number_by_date_dict, f, sort_keys=True)
```
Comment (yihong0618) [inline github_poster/loader/spotify_loader.py:67]: This two functions seems same as `bilibili` can we refactor them to one?

File: github_poster/loader/spotify_loader.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        self._parse_spotify_history()
+
+    @classmethod
+    def add_loader_arguments(cls, parser, optional):
+        parser.add_argument(
+            "--spotify_client_id",
+            dest="spotify_client_id",
+            type=str,
+            required=optional,
+            help="",
```
Comment (yihong0618) [inline github_poster/loader/spotify_loader.py:36]: some helper message?

File: github_poster/loader/spotify_loader.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        now = datetime.datetime.now()
+        yesterday = now - datetime.timedelta(
+            1,
+            hours=now.hour,
+            minutes=now.minute,
+            seconds=now.second,
+            microseconds=now.microsecond,
+        )
+        last_yesterday = yesterday + datetime.timedelta(
+            hours=23, minutes=59, seconds=59
```
Comment (yihong0618) [inline github_poster/loader/spotify_loader.py:79]: use `pendulum`
maybe better?

File: github_poster/loader/spotify_loader.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                continue
+            data_list.append(data)
+
+        next_url = track_list["next"]
+        before = int(track_list["cursors"]["before"])
+        # date_time = datetime.datetime.fromtimestamp(before//1000)
+        # print(date_time.strftime('%Y-%m-%d %H:%M:%S'))
+        if before >= after:
+            return self.get_api_data(next_url, after, limit, data_list)
+        return data_list
```
Comment (yihong0618) [inline github_poster/loader/spotify_loader.py:135]: nice code/

File: github_poster/loader/spotify_loader.py (hunk 2 above)
Comment (zjgcjy) [reply github_poster/loader/spotify_loader.py:67]: Maybe I can put it in `BaseLoader`. When needed in SpotifyLoader, use `self._type` to get the name of history file. If other loader needs the function to save history data, call it in the same way.

Comment (yihong0618) [reply github_poster/loader/spotify_loader.py:67]: its fine.

**Pull request closed without merging**